### PR TITLE
feat: first draft of the CSS Grid Magazine lesson

### LINKED ISSUE #124: Learn CSS Grid by Building a Magazine

### PR DESCRIPTION
**Pull request by thecodingaviator**
Related to the CSS Grid project

Closes #124

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
Looking good! Considering the core of this challenge is CSS grid, can you move the challenges into .css files instead of having the user write everything inside the html? The rest of the document can be distracting and we want the user to focus in on the subject at hand.

**Comment by thecodingaviator** [comment]
I'm not really sure doing that is best since we want the user to also be able to see the HTML part of what they're building because Grid does not always position elements in their order of appearance in the .html

**Comment by Ethan-Arrowood** [comment]
When we put this curriculum into action the user will be able to view the `.html`. I believe we are considering a multi-file editor experience for exactly this reason

**Comment by thecodingaviator** [comment]
Ah, okay. I was not aware of the multi-file editor thing, will separate files soon

**Comment by thecodingaviator** [comment]
@Ethan-Arrowood Does the way I'm doing it now look good?

**Comment by Ethan-Arrowood** [comment]
Yes looking good now! Is there any difference in the html files for challenges 5 or 6? If the html file isn't changing past a certain point I think it would be okay to save it as something like `index05-06.html`

**Comment by thecodingaviator** [comment]
@Ethan-Arrowood Alright, I guess the file doesn't change after 05 till 11, would make changes. Although, I'm currently importing `index07.css` in `index07.html`, but since the CSS files change, I guess I should let it be like this. What do you prefer?

**Comment by thecodingaviator** [comment]
@Ethan-Arrowood Waiting for your input to continue

**Comment by Ethan-Arrowood** [comment]
Oh I see, the import is changing. Hmm okay that is fine keep it as is for now! Depending how the platform evolves we may remove the duplicates later on.

File: learn-css-grid-by-building-a-magazine/index04.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.magazine {
+  display: grid;
+  /* To declare the number of columns i.e. vertical divisions a grid has, we use the
+  grid-template-columns property. We use a unit called fr that represents a fraction of the free space.
+  For example: grid-template-columns: 1fr 3fr; would divide the grid into 1+3=4 parts, the first column
+  would occupt 25% of the space and the second one would occupy 75%
```
Comment (Ethan-Arrowood) [inline learn-css-grid-by-building-a-magazine/index04.css]: `oocupt` -> `occupy`

File: learn-css-grid-by-building-a-magazine/index05.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.magazine {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr;
+  /* The repeat() function allows you to reduce repetition like above. It is used in the following format:
+  repeat(no of times, value to be repeated)
```
Comment (Ethan-Arrowood) [inline learn-css-grid-by-building-a-magazine/index05.css]: `no` -> `number`

File: learn-css-grid-by-building-a-magazine/index07.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
+.magazine {
+  display: grid;
+  grid-template-columns: repeat(6, 1fr);
+  /* The grid-template-areas is used to declare the layout depending on grid-area of elements. The first row of the layout has been done below,
+  make the changes to the property to:
+  Add a second row such that the first two columns belong to the element declared with the grid area of social, the next 2 with title and the last 2 with text-4
```
Comment (Ethan-Arrowood) [inline learn-css-grid-by-building-a-magazine/index07.css]: spell out the word `2` in this section. It is misleading with the `text-1` bits

**Comment by Ethan-Arrowood** [comment]
Let me know when you complete the rest of this challenge and I'll swap the labels :smile: good work so far!

File: learn-css-grid-by-building-a-magazine/index07.css (hunk 3 above)
Comment (thecodingaviator) [reply learn-css-grid-by-building-a-magazine/index07.css]: I've resolved this locally, would be pushed up with next update

File: learn-css-grid-by-building-a-magazine/index05.css (hunk 2 above)
Comment (thecodingaviator) [reply learn-css-grid-by-building-a-magazine/index05.css]: I've resolved this locally, would be pushed up with next update

File: learn-css-grid-by-building-a-magazine/index04.css (hunk 1 above)
Comment (thecodingaviator) [reply learn-css-grid-by-building-a-magazine/index04.css]: I've resolved this locally, would be pushed up with next update

**Comment by thecodingaviator** [comment]
@Ethan-Arrowood The PR is now ready for review. In the demo, I have more content, but I do not see how that would increase the knowledge of the student behind what they'll learn from what is in till `index14` so I have decided not to include it here although I'm open to opinions.

**Comment by scissorsneedfoodtoo** [comment]
@thecodingaviator, this is definitely off to a great start! I left some suggestions that would help users feel like they're building the entire project.

**Comment by scissorsneedfoodtoo** [comment]
Doesn't have to be done in this PR, just noting this for the future: it would also be really cool to see some other CSS gird concepts being taught to give the page some mobile responsiveness like this one: https://codepen.io/brianhaferkamp/pen/NpZvpx

Might be a good way to introduce things like `grid-column: 1 / span 7`.

**Comment by thecodingaviator** [comment]
I really don't have time to go ahead with breaking it down right now. Anyone interested can take this up or we can get this PR merged and create an issue which someone can take up. How does that sound @scissorsneedfoodtoo ?

**Comment by scissorsneedfoodtoo** [comment]
@thecodingaviator, sorry for the delay.

Okay, no problem! Yes, you're right -- it would be best to merge this for now and have someone else pick it up. No need to create a new issue, though. I'll just put the "Help Wanted" label back onto the issue for this project.

**Pull request merged**